playground/frontend/playground_components/lib/src/controllers/snippet_editing_controller.dart

```diff
@@ -99,9 +101,12 @@ class SnippetEditingController extends ChangeNotifier {
   /// Creates an [ExampleLoadingDescriptor] that can recover the
   /// current content.
   ExampleLoadingDescriptor getLoadingDescriptor() {
-    // TODO: Return other classes for unchanged standard examples,
-    //  user-shared examples, and an empty editor,
-    //  https://github.com/apache/beam/issues/23252
+    if (codeController.fullText.isEmpty) {
+      return EmptyExampleLoadingDescriptor(sdk: sdk);
+    }
+    if (selectedExample != null && !isChanged) {
+      return StandardExampleLoadingDescriptor(path: _selectedExample!.path);
```

Comment:
This could be other example sources. The best way to handle this is to store the original descriptor in this controller.

Reply:
fixed